oblib/parser.py

```diff
@@ -0,0 +1,216 @@
+                if fact["aspects"]["xbrl:periodStart"] == fact["aspects"]["xbrl:periodEnd"]:
+                    kwargs["instant"] = util.convert_json_datetime(fact["aspects"]["xbrl:periodStart"])
+                else:
+                    kwargs["duration"] = {}
+                    kwargs["duration"]["start"] = util.convert_json_datetime(fact["aspects"]["xbrl:periodStart"])
+                    kwargs["duration"]["end"] = util.convert_json_datetime(fact["aspects"]["xbrl:periodEnd"])
+            else:
+                kwargs["duration"] = "forever"
+
+            if "xbrl:unit" in fact["aspects"]:
+                kwargs["unit_name"] = fact["aspects"]["xbrl:unit"]
+
+            # Add axis to kwargs if item is an axis.
+            for axis_chk in fact["aspects"]:
+                if "Axis" in axis_chk:
+                    kwargs[axis_chk.split(":")[1]] = fact["aspects"][axis_chk]
+
+            entrypoint.set(fact["aspects"]["xbrl:concept"], fact["value"], **kwargs)
+
+        return entrypoint
+
+    def from_JSON(self, filename):
+        """ Imports XBRL as JSON from the given filename. """
+
+        infile = open(filename, "r")
+        s = infile.read()
+        infile.close()
+        return self.from_JSON_string(s)
+
+    def from_XML_string(self, xml_string):
+        """ Loads the Entrypoint from an XML string """
+
+        # Create an entrypoint.
+        # TODO: Should not have to initiatilize taxonomy and Entrypoint type should not be hardcoded.
+        entrypoint = data_model.Entrypoint("CutSheet", self._taxonomy)
+
+        # TODO: Supply implementation upon completion of prototype code.
+        return entrypoint
+
+    def from_XML(self, filename):
+        """ Imports XBRL as XML from the given filename. """
+
+        infile = open(filename, "r")
+        s = infile.read()
+        infile.close()
+        self.from_XML_string(s)
+
+    def to_JSON_string(self, entrypoint):
+        """ Returns XBRL as an JSON string given a data model entrypoint. """
+
+        return entrypoint.to_JSON_string()
+
+    def to_JSON(self, entrypoint, filename):
+        """ Exports XBRL as JSON to the given filename given a data model entrypoint. """
+        
+        entrypoint.to_JSON(filename)
+
+    def to_XML_string(self, entrypoint):
+        """ Returns XBRL as an XML string given a data model entrypoint. """
+
+        return entrypoint.to_XML_string()
+
+    def to_XML(self, entrypoint, filename):
+        """ Exports XBRL as XML to the given filename given a data model entrypoint. """
+        
+        entrypoint.to_XML(filename)
+
+    def convert(self, in_filename, out_filename, file_format):
+        """ 
+        Converts and input file (in_filename) to an output file (out_filename) given an input 
+        file format specified by file_format.
+        """
+
+        if file_format == FileFormat.JSON:
+            entrypoint = self.from_JSON(in_filename)
+            self.to_XML(entrypoint, out_filename)
+        else:
```

Comment:
I'd do `elif file_format.upper() == FileFormat.XML` here, and

`else:   raise ValueError ` etc.

Reply:
its an enumeration - can only have two values.

Reply:
Doesn't stop someone from trying to pass a str like I was first assuming.